frontend/components/tree/js/tree.service.ts

```diff
@@ -1092,12 +1116,14 @@ export class TreeService {
 	 * @param currentNode the node who's parents to traverse up
 	 * @param forceState a selected state to enforce upon parents
 	 */
-	public setSelectionOnParentNodes(currentNode: any, forceState?: string) {
+	public setSelectionOnParentNodes(currentNode: any, forceState?: string): any[] {
 		const parentPath = this.getPath(currentNode._id);
 		parentPath.pop(); // Remove the node itself
+		const seenParents = [];
```

Comment:
Looks like you only store this to return to L1049.

If you do the changes at L1049 then we don't need to store this anymore.